src/test/Microsoft.Extensions.DependencyModel.Tests/DependencyContextJsonReaderTest.cs

```diff
@@ -697,6 +697,20 @@ public void ReadsCompilationOptions()
             context.CompilationOptions.WarningsAsErrors.Should().Be(true);
         }
 
+        [Fact]
+        public void FailsToReadInvalidDefines()
+        {
+            Assert.Throws<FormatException>(() => Read(
```

Comment:
nit: Consider verifying that the expected line number and byte position in line are actually showing up in the message